konvoy: Update docs for konvoy release (#2802)
mesosphere/dcos-docs-site: pages/ksphere/konvoy/latest/install/install-gcp/index.md

---
layout: layout.pug
navigationTitle: Install on Google Cloud Platform (GCP)
title: Install on Google Cloud Platform (GCP)
menuWeight: 20
excerpt: Prepare for and install Konvoy on Google Cloud Platform (GCP)
enterprise: false
---

<!-- markdownlint-disable MD004 MD007 MD025 MD030 -->

> This section guides you through the basic steps to prepare your environment and install Konvoy on Google Cloud Platform (GCP).

Reviewer: ```suggestion
This section describes the steps to prepare your environment and install Konvoy on Google Cloud Platform (GCP).
```

## Prerequisites

* The [gcloud][install_gcloud] command line utility
* [Docker Desktop][install_docker] _version 18.09.2 or newer_
* [kubectl][install_kubectl] _v1.15.4 or newer_ (for interacting with the running cluster)
* A valid GCP account with [credentials configured][gcloud_sdk].
  Make sure to set your `project` and `region` you want to use and export it

  ```bash
  # set config and login with default config
  gcloud config set compute/region "YOUR_REGION"
  gcloud config set project "YOUR_PROJECT"
  gcloud auth application-default login

  # export of settings
  export GOOGLE_PROJECT=$(gcloud config get-value project)
  export GOOGLE_REGION=$(gcloud config get-value compute/region)

  # enable service APIs
  gcloud services enable cloudresourcemanager.googleapis.com
  gcloud services enable compute.googleapis.com
  gcloud services enable iam.googleapis.com
  gcloud services enable iamcredentials.googleapis.com
  gcloud services enable stackdriver.googleapis.com
  gcloud services enable pubsub.googleapis.com
  ```

>   You need to have at least the following permissions within your GCP project.

Reviewer: ```suggestion
<p class="message--note"><strong>NOTE: </strong>You must have, as a minimum, the following permissions in your GCP project.</p>
```

- compute.targetPools.addInstance
- compute.targetPools.create
- compute.targetPools.delete
- compute.targetPools.get
- compute.targetPools.list
- compute.targetPools.removeHealthCheck
- compute.targetPools.update
- compute.targetPools.use
- compute.targetTcpProxies.create
- compute.targetTcpProxies.delete
- compute.targetTcpProxies.get
- compute.targetTcpProxies.list
- compute.targetTcpProxies.update
- compute.targetTcpProxies.use
- compute.zoneOperations.get
- compute.zones.get
- compute.zones.list
- iam.roles.create
- iam.roles.delete
- iam.roles.get
- iam.roles.list
- iam.roles.update
- iam.serviceAccountKeys.create
- iam.serviceAccountKeys.delete
- iam.serviceAccountKeys.list
- iam.serviceAccounts.actAs
- iam.serviceAccounts.create
- iam.serviceAccounts.delete
- iam.serviceAccounts.get
- iam.serviceAccounts.getIamPolicy
- iam.serviceAccounts.list
- iam.serviceAccounts.setIamPolicy
- iam.serviceAccounts.update
- orgpolicy.policy.get
- pubsub.topics.create
- resourcemanager.projects.get
- resourcemanager.projects.getIamPolicy
- resourcemanager.projects.setIamPolicy
- resourcemanager.projects.update
- servicemanagement.services.bind
- servicemanagement.services.update
```

> Command to create a role via GCP SDK.

Reviewer: ```suggestion
# Create a role using the GCP SDK commands
```

> **Just create or update it with the command, if you do not have the role in your project**

Reviewer: ```suggestion
<p class="message--note"><strong>NOTE: </strong>Create or update the role with the command, if the role is not in your project</p>
```

```bash
gcloud iam roles create konvoy --project ${GOOGLE_PROJECT} --file ./konvoy-role.yaml
```

> Commands to create a service account for `konvoy` deployment and use it for spinning up a Kubernetes cluster.

Reviewer: I don't know if this was intended to be a section head or not
```suggestion
# Create a service account for `konvoy` deployment and for spinning up a Kubernetes cluster.
```

> You can do this if you do not want to use your own user credentials, you [logged in][gcloud_sdk] with.

Reviewer: ```suggestion
You can do this if you do not want to use your own user [logged in][gcloud_sdk] credentials.
```

```bash
gcloud iam service-accounts create konvoy --description "Management of Konvoy clusters" --display-name "Konvoy Management"
gcloud projects add-iam-policy-binding ${GOOGLE_PROJECT} --member serviceAccount:konvoy@${GOOGLE_PROJECT}.iam.gserviceaccount.com --role projects/${GOOGLE_PROJECT}/roles/konvoy
gcloud iam service-accounts keys create ${HOME}/.config/gcloud/konvoy-sa-${GOOGLE_PROJECT}.json --iam-account konvoy@${GOOGLE_PROJECT}.iam.gserviceaccount.com
export GOOGLE_APPLICATION_CREDENTIALS="${HOME}/.config/gcloud/konvoy-sa-${GOOGLE_PROJECT}.json"
```

## Installation

> After verifying your prerequisites, you can create an GCP Kubernetes cluster by running `konvoy up --provisioner gcp`.

Reviewer: ```suggestion
After verifying your prerequisites, create an GCP Kubernetes cluster by running `konvoy up --provisioner gcp`.
```

This command creates your GCP [VM instances][compute_virtual_machine], installs Kubernetes and default add-ons to support your Kubernetes cluster.

Specifically, the `konvoy up --provisioner gcp` command does the following:

* Provisions three `n1-standard-4` virtual machines as Kubernetes master nodes
* Provisions four `n1-standard-8` virtual machines as Kubernetes worker nodes
* Deploys all of the following default add-ons:
  * Calico
  * Cert-Manager
  * CoreDNS
  * Helm
  * GCE Disk CSI driver
  * Elasticsearch (including Elasticsearch Exporter)
  * Fluent Bit
  * Kibana
  * Prometheus operator (including Grafana, AlertManager and Prometheus Adapter)
  * Traefik
  * Kubernetes dashboard
  * Operations portal
  * Velero
  * Dex identity service
  * Dex Kubernetes client authenticator
  * Traefik forward authorization proxy
  * Kommander
  * Reloader
  * Default Storage Class Protection
  * Gatekeeper
  * Kudo
  * Konvoy Config

The default configuration options are recommended for a small cluster (about 10 worker nodes).

## Modifying the cluster name

> By default, the cluster name is the name of the folder where your run the `konvoy` command.

Reviewer: ```suggestion
By default, the cluster name is the name of the folder where the konvoy command is run. 
```

> The cluster name will be used to tag the provisioned infrastructure and the context when applying the kubeconfig file.

Reviewer: ```suggestion
The cluster name is used to tag the provisioned infrastructure and the context when applying the kubeconfig file.
```

To customize the cluster name, run the following command:

```bash
konvoy up --provisioner gcp --cluster-name <YOUR_SPECIFIED_NAME>
```

## Control plane and worker nodes

> Control plane nodes are the nodes where the Kubernetes Control Plane components will be installed.

Reviewer: ```suggestion
Control plane nodes are the nodes where the Kubernetes Control Plane components are installed.
```

> The Control Plane contains [various components][control_plane], including `etcd`, `kube-apiserver` (that [you will interact with through `kubectl`][kubectl]), `kube-scheduler` and `kube-controller-manager`. Please also refer to the [Concepts section][concepts].

Reviewer: ```suggestion
The Control Plane contains [various components][control_plane], including `etcd`, `kube-apiserver` (that [you will interact with through `kubectl`][kubectl]), `kube-scheduler` and `kube-controller-manager`. Refer to the [Concepts section][concepts] for more information.
```

Having three control plane nodes makes the cluster "highly available" to protect against failures.
Worker nodes run your containers in [Kubernetes pods][pods].

## Default addons

The default addons help you manage your Kubernetes cluster by providing monitoring (Prometheus), logging (Elasticsearch), dashboards (Kubernetes Dashboard), storage (GCE Disk CSI Driver), ingress (Traefik) and other services.

## Viewing installation operations

> As noted above, you start the cluster installation by running the `konvoy up --provisioner gcp` command.

Reviewer: ```suggestion

```

> As the `konvoy up --provisioner gcp` command runs, you will see output about the operations performed.

Reviewer: ```suggestion
As the `konvoy up --provisioner gcp` command runs, to start the cluster installation, you will see output as the operations are performed.
```

> The first set of messages you see is the output generated by [Terraform][terraform] as it provisions your nodes.

Reviewer: ```suggestion
The first output messages you see are from [Terraform][terraform] as it provisions your nodes.
```

After the nodes are provisioned, [Ansible][ansible] connects to the instances and installs Kubernetes in steps called tasks and playbooks.
Near the end of the output, addons are installed.

## Viewing cluster operations

> You can access user interfaces to monitor your cluster through the [Operations Portal][ops_portal].

Reviewer: ```suggestion
You can monitor your cluster through the [Operations Portal][ops_portal] user interface.
```
